### DISCUSSION TITLE
Cannot purge RabbitMQ queues with pika

### DISCUSSION THREAD

**Discussion by Montana** [pika/pika#1395] [Q&A]
Hi all,

Producer code: 

```python
conn_params = pika.ConnectionParameters('localhost', 5672,  '/')
connection = pika.BlockingConnection(conn_params)
channel = connection.channel()
channel.exchange_declare(exchange='direct_exchange',
                         exchange_type='direct')
channel.queue_declare(queue='my_queue')

@socketIo.on("clean")
def clean(msg):
    print("cleaning")
    channel.queue_purge('my_queue')
```
Consumer code: 

```python
conn_params = pika.ConnectionParameters('localhost', 5672, '/')
connection = pika.BlockingConnection(conn_params)
channel = connection.channel()
channel.exchange_declare(exchange='direct_exchange', exchange_type='direct')
result = channel.queue_declare(queue='my_queue')
channel.queue_bind(exchange='direct_exchange', queue='my_queue', routing_key='my_queue')


def callback(ch, method, properties, body):
    ...
    channel.basic_ack(delivery_tag=method.delivery_tag)

channel.basic_consume('my_queue', callback, auto_ack=False )
channel.basic_qos(prefetch_count=1)
```
I've tuned the variables `rabbit_host` and `rabbit_queue` according to my needs. The problem is `pika` always shows `0` both in the producer and performer.

**Comment by lukebakken** [comment]
> The problem is pika always shows 0 both in the producer and performer

It's not clear what that statement means.

The best way to get assistance is to provide something I can run to see the issue, rather than code snippets, because the latter requires that I figure out how to reproduce the issue.

I have some time to come up with something to fill a queue and then use Pika to purge it. Please follow this repo - https://github.com/lukebakken/pika-1395

**Reply by lukebakken** [reply]
@Montana You can run the above example via:

```
docker compose up --build
```

Everything appears to run as expected.
